livekit-ffi/protocol/audio_frame.proto

```diff
@@ -113,14 +113,19 @@ message OwnedAudioStream {
 }
 
 message AudioStreamEvent {
-  uint64 source_handle = 1;
-  oneof message { AudioFrameReceived frame_received = 2; }
+  uint64 stream_handle = 1;
+  oneof message { 
+    AudioFrameReceived frame_received = 2;
+    AudioStreamEOS eos = 3;
```

Comment:
this works, but you can also consider to use a bool as a marker. with oneof messages, the position/field name is sufficient to mark special types.

Reply:
TIL, I think it's ok for now